frontstage_api/controllers/eq_payload.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import logging
+import time
+import uuid
+
+from flask import current_app
+from frontstage_api.controllers import case_controller, collection_exercise_controller, \
+    collection_instrument_controller, party_controller, survey_controller
+from frontstage_api.exceptions.exceptions import InvalidEqPayLoad
+from structlog import wrap_logger
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class EqPayload(object):
+
+    def create_payload(self, case_id):
+        """
+        Creates the payload needed to communicate with EQ, built from the Case, Collection Exercise, Party,
+        Survey and Collection Instrument services
+        :case_id: The unique UUID references of a case
+        :return Payload for EQ
+        """
+
+        tx_id = str(uuid.uuid4())
+        logger.info('Creating payload for JWT', case_id=case_id, tx_id=tx_id)
+
+        case = case_controller.get_case_by_case_id(case_id)
+
+        # Collection Instrument
+        ci_id = case['collectionInstrumentId']
+        ci = collection_instrument_controller.get_collection_instrument(ci_id)
+        if ci['type'] != 'EQ':
+            raise InvalidEqPayLoad('Collection instrument {} type is not EQ'.format(ci_id))
+
+        classifiers = ci['classifiers']
+        if not classifiers or not classifiers.get('EQ_ID') or not classifiers.get('FORM_TYPE'):
+            raise InvalidEqPayLoad('Collection instrument {} classifiers are incorrect or missing'.format(ci_id))
+
+        eq_id = ci['classifiers']['EQ_ID']
+        form_type = ci['classifiers']['FORM_TYPE']
+
+        # Collection Exercise
+        collex_id = case["caseGroup"]["collectionExerciseId"]
+        collex = collection_exercise_controller.get_collection_exercise(collex_id)
+        collex_event_dates = self._get_collex_event_dates(collex_id)
+
+        # Party
+        party_id = case['caseGroup']['partyId']
+        party = party_controller.get_party_by_business_id(party_id, collection_exercise_id=collex_id)
+
+        # Survey
+        survey_id = collex['surveyId']
+        survey = survey_controller.get_survey(survey_id)
+
+        account_service_url = current_app.config['ACCOUNT_SERVICE_URL']
+        iat = time.time()
+        exp = time.time() + (5 * 60)
```

Comment:
+ datetime.timedelta(minutes=5) might be more readable. Where is it defined that the expiry is 5 minutes? That seems quite long?

Reply:
Thats what has been agreed, just matched the other implementations